### PR TITLE
suppress use of ROL for older versions of Trilinos

### LINKED ISSUE #5456: rol tests failing
ROL in Trilinos are failing on this tester https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&buildid=13149 

with errors
```
In file included from /mnt/data/testsuite/dealii/tests/rol/vector_adaptor_01.cc:21:0:
/mnt/data/testsuite/dealii/include/deal.II/optimization/rol/vector_adaptor.h:226:33: error: ‘Elementwise’ in namespace ‘ROL’ does not name a type
     void applyUnary (const ROL::Elementwise::UnaryFunction<value_type> &f);
     
```

this is with `TRILINOS_VERSION: 12.2.1`. 

Probably the interfaces in ROL changed we need to require some minimum version of Trilinos. (certainly works with  `12.12.1`).

@vishalkenchan could you please try to figure out which version of Trilinos has the interface to ROL you expect?

### PR DESCRIPTION
**Pull request by vishalkenchan**
To fix #5456

### CODE REVIEW

File: cmake/configure/configure_2_trilinos.cmake [hunk 1]
```diff
@@ -66,6 +66,15 @@ MACRO(FEATURE_TRILINOS_FIND_EXTERNAL var)
       ENDIF()
     ENDFOREACH()
 
+    IF(${DEAL_II_TRILINOS_WITH_ROL})
+      IF(TRILINOS_VERSION VERSION_LESS 12.6.1)
+        MESSAGE(STATUS "ROL interface is disabled."
+          "deal.II will not support ROL interface if Trilinos version is less "
+          "than 12.6.1. Trilinos version found: \"${TRILINOS_VERSION}\".")
```
Comment (masterleinad) [inline cmake/configure/configure_2_trilinos.cmake]: Do you know if this is sharp?

**Comment by davydden** [comment]
./run-tests

File: cmake/configure/configure_2_trilinos.cmake (hunk 1 above)
Comment (vishalkenchan) [reply cmake/configure/configure_2_trilinos.cmake]: I am not sure about Trilinos@12.4.1 as I am not able to build it without errors so far. But Trilinos@12.6.1. But if someone could improve the sharpness.

Comment (bangerth) [reply cmake/configure/configure_2_trilinos.cmake]: Let's not worry too much about sharpness. The people who have old versions of Trilinos already built did so likely without ROL. If someone wants to use ROL, they likely have to rebuild things, and then they may as well use the latest version.

**Comment by bangerth** [comment]
By the way, did you update the Trilinos build instructions in `doc/` to make people build ROL as well? We should just generally recommend that they do so -- no harm done if someone doesn't use it, but it makes the instructions simpler.

**Comment by vishalkenchan** [comment]
@bangerth I see. I wasn't aware of this. I will do it in this PR.

**Comment by bangerth** [comment]
OK to merge once the tester is happy.

**Comment by bangerth** [comment]
Thanks, @vishalkenchan. In general, use separate pull requests for separate things -- that makes it easier to review things, and easier to keep separate things apart. But it was a small change here, so not worth fussing over. Thanks for fixing the issue in question here!

**Pull request merged**